drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -2402,6 +2420,22 @@ static int rswitch_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 		goto out;
 	}
 
+	if (rdev->is_vmq) {
+		if (rswitch_is_front_dev(rdev)) {
+			rx_chain_free = READ_ONCE(rdev->vmq_info->rx_back_ring_size) -
+					(READ_ONCE(rdev->vmq_info->front_tx) -
+					READ_ONCE(rdev->vmq_info->back_rx));
+		} else {
+			rx_chain_free = READ_ONCE(rdev->vmq_info->rx_front_ring_size) -
+					(READ_ONCE(rdev->vmq_info->back_tx) -
+					READ_ONCE(rdev->vmq_info->front_rx));
+		}
+		if (num_desc > rx_chain_free) {
+			ret = NETDEV_TX_BUSY;
```

Comment:
Should we stop the queue in this case? I believe that `NETDEV_TX_BUSY` is the hard error and we really need to avoid signaling it.

Reply:
Good point! Added netif_stop_subqueue in the latest PR version.

Reply:
AFAIK, we need another check that the subqueue when we use last free descriptor. Again, returning `NETDEV_TX_BUSY` is the last resort.

Check other network drivers,  they are stopping subqueue earlier.